addOns/grpc/src/main/java/org/zaproxy/addon/grpc/internal/EncoderUtils.java

```diff
@@ -0,0 +1,339 @@
+            if (first == -1 || last == -1)
+                throw new InvalidProtobufFormatException(
+                        Constant.messages.getString("grpc.encoder.nested.message.newline.error"));
+            return text.substring(first + 1, last);
+        } else {
+            throw new InvalidProtobufFormatException(
+                    Constant.messages.getString("grpc.encoder.nested.message.braces.error"));
+        }
+    }
+
+    public static List<String> parseIntoList(String inputString)
+            throws InvalidProtobufFormatException {
+        if (inputString == null || inputString.isEmpty()) {
+            return new ArrayList<>();
+        }
+        List<String> output = new ArrayList<>();
+        String[] temp = inputString.split("\n");
+
+        StringBuilder field = new StringBuilder();
+        int countOpenCurlyBraces = 0;
+
+        for (String s : temp) {
+            if (s.charAt(s.length() - 1) != '{'
+                    && s.charAt(0) != '}'
+                    && countOpenCurlyBraces == 0) {
+                output.add(s);
+            } else if (s.charAt(s.length() - 1) == '{') {
+                countOpenCurlyBraces++;
+                field.append(s).append('\n');
+            } else if (s.charAt(0) == '}') {
+                countOpenCurlyBraces--;
+                if (countOpenCurlyBraces == 0) {
+                    field.append(s);
+                    output.add(field.toString());
+                    field.setLength(0);
+                } else {
+                    field.append(s).append('\n');
+                }
+            } else {
+                field.append(s).append('\n');
+            }
+        }
+
+        if (countOpenCurlyBraces != 0) {
+            output.clear();
+            throw new InvalidProtobufFormatException(
+                    Constant.messages.getString(
+                            "grpc.encoder.nested.message.missing.braces.error"));
+        }
+        return output;
+    }
+
+    public static byte[] hexStringToByteArray(String hexString) {
+        // Ensure even length
+        if (hexString.length() % 2 != 0) {
+            throw new IllegalArgumentException("Invalid Hex string: Odd length");
+        }
+
+        // Create byte array with correct size
+        byte[] result = new byte[hexString.length() / 2];
+
+        // Parse hex pairs and fill the byte array
+        for (int i = 0; i < result.length; i++) {
+            String hexByte = hexString.substring(i * 2, i * 2 + 2);
+            int byteValue = Integer.parseInt(hexByte, 16); // Parse as base-16 (hex)
+            result[i] = (byte) byteValue;
+        }
+
+        return result;
+    }
+
+    public static String removeDoubleQuotes(String str) throws InvalidProtobufFormatException {
+        // Check if the string starts and ends with a double quote
+        if (str.startsWith("\"") && str.endsWith("\"")) {
+            // Remove the first and last characters
```

Comment:
These comments are not needed, they are just repeating the code, better comments say why (if really needed).